[7_1_X][TIMOB-9366] Android: Added "radial" gradient support

jquick-axway opened the pull request:
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-9366

**Cherry Pick of:** PR #9815

jquick-axway committed:
[TIMOB-9366] Android: Added "radial" gradient support to view's "backgroundGradient" property

hansemannn committed:
Update TiGradientDrawable.java

jquick-axway committed:
Android: Updated [TIMOB-9366] code based on feedback.

- Also modified "TiUIView" to log exception thrown by "TiGradientDrawable" constructor. Particularly for invalid "type" parameters.

jquick-axway committed:
Android: Fixed radial gradient crash caused by setting radius to zero.

jquick-axway committed:
Merge branch '7_1_X' into TIMOB-9366-7_1_X

build commented:
<!--
  1 failure:  :disappointed_rel...
  0 warning: 
  1 messages
  
-->

<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  :disappointed_relieved: `npm test` failed. See below for details.
  </td>
    </tr>
  </tbody>
</table>



<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>

  ```
> titanium-mobile@7.1.0 test /Users/build/jenkins/workspace/sdk_titanium_mobile_PR-9855-2RWTRBIJ3G2QYECPBHI5KZSH3AWCJBF6LFFQBTP6W2E5IZHICOSA
> grunt

Running "appcJs:src" (appcJs) task

Running "eslint:src" (eslint) task

/Users/build/jenkins/workspace/sdk_titanium_mobile_PR-9855-2RWTRBIJ3G2QYECPBHI5KZSH3AWCJBF6LFFQBTP6W2E5IZHICOSA/android/cli/commands/_buildModule.js
  554:18  warning  Found non-literal argument to RegExp Constructor  security/detect-non-literal-regexp

/Users/build/jenkins/workspace/sdk_titanium_mobile_PR-9855-2RWTRBIJ3G2QYECPBHI5KZSH3AWCJBF6LFFQBTP6W2E5IZHICOSA/tests/Resources/ti.ui.picker.addontest.js
   85:5  warning  Expected an assignment or function call and instead saw an expression  no-unused-expressions
  122:5  warning  Expected an assignment or function call and instead saw an expression  no-unused-expressions
  123:5  warning  Expected an assignment or function call and instead saw an expression  no-unused-expressions
  171:5  warning  Expected an assignment or function call and instead saw an expression  no-unused-expressions
  172:5  warning  Expected an assignment or function call and instead saw an expression  no-unused-expressions
  175:5  warning  Expected an assignment or function call and instead saw an expression  no-unused-expressions
  176:5  warning  Expected an assignment or function call and instead saw an expression  no-unused-expressions
  245:5  warning  Expected an assignment or function call and instead saw an expression  no-unused-expressions
  246:5  warning  Expected an assignment or function call and instead saw an expression  no-unused-expressions
  252:5  warning  Expected an assignment or function call and instead saw an expression  no-unused-expressions
  253:5  warning  Expected an assignment or function call and instead saw an expression  no-unused-expressions

✖ 12 problems (0 errors, 12 warnings)


Running "continue:on" task

Running "retire:js" (retire) task
Ignoring {"paths":["growl"],"descriptors":[]}
Downloading https://raw.github.com/RetireJS/retire.js/master/repository/jsrepository.json ...
(node:55669) [DEP0013] DeprecationWarning: Calling an asynchronous function without callback is deprecated.
Checking: dangerfile.js
Checking: Gruntfile.js
Checking: apidoc/docgen.js
Checking: apidoc/lib/addon_generator.js
Checking: apidoc/lib/changes_generator.js
Checking: apidoc/lib/common.js
Checking: apidoc/lib/html_generator.js
Checking: apidoc/lib/jsca_generator.js
Checking: apidoc/lib/jsduck_generator.js
Checking: apidoc/lib/json_generator.js
Checking: apidoc/lib/modulehtml_generator.js
Checking: apidoc/lib/parity_generator.js
Checking: apidoc/lib/solr_generator.js
Checking: apidoc/validate.js
Checking: build/android.js
Checking: build/androidsdk.js
Checking: build/ant.js
Checking: build/docs.js
Checking: build/git.js
Checking: build/ios.js
Checking: build/packager.js
Checking: build/scons-build.js
Checking: build/scons-clean.js
Checking: build/scons-cleanbuild.js
Checking: build/scons-install.js
Checking: build/scons-modules-integrity.js
Checking: build/scons-package.js
Checking: build/scons-ssri.js
Checking: build/scons-test.js
Checking: build/scons-update-node-deps.js
Checking: build/scons.js
Checking: build/titanium-sdk/lib/titanium.js
Checking: build/utils.js
Checking: build/windows.js
Checking: cli/commands/build.js
Checking: cli/commands/clean.js
Checking: cli/commands/create.js
Checking: cli/commands/project.js
Checking: cli/hooks/hyperloop-fix.js
Checking: cli/lib/creator.js
Checking: cli/lib/creators/app.js
Checking: cli/lib/creators/applewatch.js
Checking: cli/lib/creators/module.js
Checking: cli/lib/info.js
Checking: android/cli/commands/_build.js
Checking: android/cli/commands/_buildModule.js
Checking: android/cli/hooks/aar-transform.js
Checking: android/cli/hooks/package.js
Checking: android/cli/hooks/run.js
Checking: android/cli/lib/AndroidManifest.js
Checking: android/cli/lib/base-builder.js
Checking: android/cli/lib/detect.js
Checking: android/cli/lib/info.js
Checking: android/cli/tests/run.js
Checking: android/cli/tests/test-androidmanifest.js
Checking: iphone/cli/commands/_build.js
Checking: iphone/cli/commands/_buildModule.js
Checking: iphone/cli/hooks/frameworks.js
Checking: iphone/cli/hooks/hyperloop.js
Checking: iphone/cli/hooks/install.js
Checking: iphone/cli/hooks/package.js
Checking: iphone/cli/hooks/run.js
Checking: iphone/cli/lib/info.js
Checking: tests/Resources/ti.example.addontest.js
Checking: tests/Resources/ti.ui.picker.addontest.js
Checking: tests/Resources/ti.ui.tableview.addontest.js
No vulnerabilities found.

Running "retire:node" (retire) task
Ignoring {"paths":["growl"],"descriptors":[]}
Downloading https://raw.github.com/RetireJS/retire.js/master/repository/npmrepository.json ...
(node:55669) [DEP0013] DeprecationWarning: Calling an asynchronous function without callback is deprecated.
Checking: .
Missing version for babel-eslint. Need to run npm install ?
Missing version for eslint-plugin-jsx-a11y. Need to run npm install ?
Missing version for eslint-plugin-react. Need to run npm install ?
Missing version for babel-eslint. Need to run npm install ?
Missing version for eslint-plugin-jsx-a11y. Need to run npm install ?
Missing version for eslint-plugin-react. Need to run npm install ?
No vulnerabilities found.

Running "continue:off" task

Running "ios_format:src" (ios_format) task

Running "android_format:src" (android_format) task
Fatal error: Error: Formatting incorrect on "android/titanium/src/java/org/appcelerator/titanium/view/TiGradientDrawable.java", proposed changes: <?xml version='1.0'?>
<replacements xml:space='preserve' incomplete_format='false'>
<replacement offset='14426' length='8'></replacement>
<replacement offset='14492' length='1'>&#10;													</replacement>
</replacements>

npm ERR! Test failed.  See above for more details.

```
  </td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

jquick-axway committed:
Android: Updated code formatting for [TIMOB-9366]

jquick-axway committed:
Android: Made additional radial gradient changes for [TIMOB-9366]

- Fixed bug where if start radius is zero, a single transparent pixel can be seen in center of circle.
- Fixed bug where Android 4.2-4.4 shows glitchy graphics when start radius is greater than 0.0 and backfillStart is false.
  * This was a Google bug that had to be worked-around by adding a color offset starting from 0.0 (center of circle).
- Fixed bug where outer edge of circle would wrongly back-fill if radius was greater than 255 and if border was applied to view (HW acceleration disabled) or GPU shader is limited to "lowp" 1 byte floats.

garymathews reviewed (approved):
CR: **PASS** once updated from `master` PR

lokeshchdhry commented:
FR Passed.

Used the test code in the ticket & `Start Radius`, ``End Radius`, `Back Fill End` & `Back Fill Start` works as expected in various scenarios.

Studio Ver:     5.0.0.201712081732
SDK Ver:        7.1.0 local build
OS Ver:         10.13.2
Xcode Ver:      Xcode 9.2
Appc NPM:       4.2.12
Appc CLI:       7.0.2
Daemon Ver:     1.0.1
Ti CLI Ver:     5.0.14
Alloy Ver:      1.11.0
Node Ver:       8.9.1
NPM Ver:        5.5.1
Java Ver:       1.8.0_101
Devices:        ⇨ google Nexus 6P --- Android 8.0.0
                ⇨ google Nexus 5 --- Android 6.0.1
Emulator: Android 4.1, 4.2.2

Commit:
Merge branch '7_1_X' into TIMOB-9366-7_1_X
